std/typecons.d

```diff
@@ -2387,99 +2387,257 @@ if (is(T == class) || is(T == interface) || isDynamicArray!T || isAssociativeArr
     a = new Widget;
 }
 
-// https://issues.dlang.org/show_bug.cgi?id=16054
-@safe unittest
+/// ditto
+struct Rebindable(T)
+if (!is(T == class) && !is(T == interface) && !isDynamicArray!T && !isAssociativeArray!T)
```

Comment:
Are we really targeting all types? Or is this another way to say `T == struct || T == basic type` ? T could also be an AliasSeq, is that supposed to be supported?

Reply:
`T` cannot be an `AliasSeq` because `AliasSeq` doesn't match to `T`. Everything else is targeted, sure. What wouldn't be?

Reply:
Isn't the condition cleaner and more expressive if it states the targeted types rather than negating the ones that are not?

Reply:
Right, but I genuinely don't know which types it wouldn't work for, so the condition would just be a list of "every type in D except those handled by the other Rebindable implementation." Which is better expressed by inverting the condition from that type.